test/test_grid.py

```diff
@@ -52,16 +52,147 @@ def test_open_non_mesh2_write_exodus(self):
         grid.encode_as("exodus")
 
     def test_init_verts(self):
```

Comment:
This is a very comprehensive test case, great work!

Can you create another case to see what happens if the user passes through an array/list/tuple with invalid dimensions, especially if only a small part of the input is invalid, such as the following

```Python
faces_verts_tuples = [
            ((150, 10, 12345678), (-1,-1), (150, 30), (135, 30), (125, 20), (135, 10)),
            ((125, 20), (135, 30), (125, 60), (110, 60), (100, 30), (105, 20)),
            ((95, 10), (105, 20), (100, 30), (85, 30), (75, 20, 1234, 4321), (85, 10))
        ]

Reply:
The initial `Grid.__from_vert()__` doesn't do any values filtering, so as this expanded `Grid.__from_vert()__`. The `(-1,-1)` will cause the `        unique_verts, indices = np.unique(dataset.reshape(-1, dataset.shape[-1]),axis=0,return_inverse=True)` to throw an error.

I also noticed that the original `Grid.__from_vert__()` couldn't handle the input vertices with `Fill_values` as well.

I think for this PR, we can only focus on the multiple-face cases, and create a new PR for the data screening

Reply:
That's fine with me